xdesign/propagation.py

```diff
@@ -192,6 +173,7 @@ def far_propagate(grid, wavefront, dist_nm):
 def _far_propagate_2(grid, wavefront, lmd, z_um):
     """Free space propagation using product Fourier algorithm.
     """
+    raise DeprecationWarning
```

Comment:
If you `raise` a warning instead of using `warnings.warn()`, it will exit the program instead of just displaying the warning.